tests/foreman/api/test_subnet.py

```diff
@@ -0,0 +1,399 @@
+
+        :id: c1dae6f4-45b1-45db-8529-d7918e41a99b
+
+        :steps:
+
+            1. Create Subnet with all the details
+            2. Create subnet parameter with single key and single value
+
+        :expectedresults: The parameter should be created in subnet
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_create_with_parameter_and_multiple_values(self):
+        """Subnet parameters can be created with multiple values
+
+        :id: b3de6f96-7c39-4c44-b91c-a6d141f5dd6a
+
+        :steps:
+
+            1. Create Subnet with all the details
+            2. Create subnet parameter having single key and multiple values
+                separated with comma
+
+        :expectedresults: The parameter with multiple values should be saved
+            in subnet
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_create_with_parameter_and_multiple_names(self):
+        """Subnet parameters can be created with multiple names with valid
+        separators
+
+        :id: d1e2d75a-a1e8-4767-93f1-0bb1b75e10a0
+
+        :steps:
+
+            1. Create Subnet with all the details
+            2. Create subnet parameter having key with multiple names separated
+                by valid separators(e.g fwd slash) and value
+
+        :expectedresults: The parameter with multiple names separated by valid
+            separators should be saved in subnet
+        """
+
+    @stubbed
+    @tier1
+    def test_negative_create_with_parameter_and_invalid_separator(self):
+        """Subnet parameters can not be created with multiple names with
+        invalid separators
+
+        :id: 08d10b75-a0db-4a11-a915-965a2a207d16
+
+        :steps:
+
+            1. Create Subnet with all the details
+            2. Create subnet parameter having key with multiple names separated
+                by invalid separators(e.g spaces) and value
+
+        :expectedresults: The parameter with multiple names separated by
+            invalid separators should not be saved in subnet
+        """
+
+    @stubbed
+    @tier1
+    def test_negative_create_with_duplicated_parameters(self):
+        """Duplicate parameters cannot be created in Subnet
+
+        :id: aa69bdcc-e833-41e4-8f72-7139bdd64daa
+
+        :steps:
+
+            1. Create Subnet with all the details
+            2. Create Multiple parameters having duplicate key names
+
+        :expectedresults: The subnet parameters should not be created with
+            duplicate names
+        """
+
+    @stubbed
+    @skip_if_bug_open('bugzilla', 1470014)
+    @tier3
+    def test_positive_inherit_subnet_parmeters_in_host(self):
+        """Host inherits parameters from subnet
+
+        :id: e372a594-f758-40ef-95ec-867937e44b63
+
+        :steps:
+
+            1. Create valid subnet with a valid parameter
+            2. Create host with above subnet
+            3. Assign hosts primary interface with subnet
+            4. List inherited subnet parameters in above host
+
```

Comment:
Let's me briefly explain the paradigm.
As you can, on the start of you module, you have next text:
:Requirement: Subnet

:CaseAutomation: Automated

*:CaseLevel: Acceptance*

:CaseComponent: API

:TestType: Functional

:CaseImportance: High

That let's you to not specify CaseLevel for tier1 as you already specify it on a module level. In case you want to introduce any other tier level, you have to override that value

for tier2:
`CaseLevel: Integration`

for tier3:
`CaseLevel: System`

for tier4:
same `CaseLevel: System` as tier4 is not real tier level, but one that we introduce just to mark tests that run for long time